src/webui/www/private/setlocation.html

```diff
@@ -31,7 +31,7 @@
                 new Event(e).stop();
                 // check field
                 var location = $('setLocation').value.trim();
-                if (location === null || location === "")
+                if (location === null)
```

Comment:
The empty string check looks like it prevented the form from being submitted when the location was blank. Is there some reason we'd want it gone?

Reply:
Now the server checks if the field is blank. This allows to show error messages.
![image](https://user-images.githubusercontent.com/4264235/41524153-32304b9c-72e5-11e8-81e2-6a3434746133.png)

Reply:
We should still perform this check in the client since it can save an extra network request. The server side check should still exist though, for robustness.

Reply:
In this case, we need to duplicate the error message and complicate localization, or leave the user without the message at all.

Reply:
The message text will be duplicated between the html file and the controller- I don't see a way around that- but the localization need only occur once. You can use the `torrentscontroller.cpp` localized string by specifying `QBT_TR(Save path is empty)QBT_TR[CONTEXT=TorrentsController]` in your html.